### ISSUE TITLE
Bug: two blank lines will be inserted between headings and tags if enable heading-blank-lines with Bottom=true

### ISSUE THREAD

**Issue by XDcedar** [platers/obsidian-linter#501]
## Describe the Bug

Hi, I usually write diary in this format:

```markdown
## diary

#study
blah blah blah

#life
blah blah blah
```

And after linting, there would be two blank lines between `#diary` and `#study`.

## How to Reproduce

1. Enable the option `heading-blank-lines` and `Bottom`.
2. Create a note like above.
3. Lint the file
4. See two blank lines between heading and tag.

## Expected Behavior

Same as input.

## Device

- [x] Desktop
- [x] Mobile

**Comment by pjkaufman** [comment]
Thanks for reporting this @XDcedar.

Do you have any other blank line rules active or is the heading blank lines the only one? I have not tested this myself yet, but I want to make sure there are no other possible culprits.

**Comment by pjkaufman** [comment]
I believe I know what the issue is with this, but I need to go ahead and verify that the solution works. I believe that the tag regex is accidentally hiding the last new line character which causes it to get inserted. I need to try a custom function and see if that handles this scenario.

**Comment by XDcedar** [comment]
Hi @pjkaufman , I'm pretty sure. This issue appears when I enable the `Bottom` option. And it disappears when I disable the `Bottom` option.
Furthermore, I made a copy of my vault for testing just now. Here is what I found:
- Enable every rule in Spacing tab, and lint the file. This issue disappears.
- Disable rule [`paragraph-blank-lines`](https://github.com/platers/obsidian-linter/blob/master/docs/rules.md#paragraph-blank-lines) (This is what I usually do). This issue appears.
- Disable rule `Bottom` in `heading-blank-lines`. This issue disappears again.

So it shows that `heading-blank-lines` with `Bottom=True` introduces this issue, and `paragraph-blank-lines` fixes it by accident (by removing those two consecutive blank lines, I guess).
